### DISCUSSION TITLE
KMS decrypt using CustomSMSSender Lambda

### DISCUSSION THREAD

**Discussion by sudsbangs** [aws/aws-sdk-js-v3#3691] [Q&A] [answered]
Hi,

I am trying to add a custom sms lambda trigger in cognito which would decrypt the code sent by cognito trigger when MFA has been enabled for user using KMS symmetric key. However i encounter the following issue `Invoke Error {"errorType":"InvalidCiphertextException","errorMessage":"UnknownError"` `@aws-sdk/client-kms'`. There are no sufficient documentation on how do we decrypt using v3 when the cognito sends an event to CustomSMSSender lambda.  

Please see my sample code below: 
`
export const customSMSSender = async (
  event: any,
  context: APIGatewayEventRequestContext,
  callback: Callback
) => {`
 ` try { `
    `const decryptInputParams: DecryptCommandInput = {`
    `  CiphertextBlob: Buffer.from(event.request.code, 'base64'),`
   ` };`
    `const decryptCommand: DecryptCommand = new DecryptCommand(
      decryptInputParams
    );`
    `const decryptRes = await kmsClient.send(decryptCommand);`
`} catch (e) {`
    // Error thrown as below `
`{
    `"errorType": "InvalidCiphertextException",
    "errorMessage": "UnknownError",
    "name": "InvalidCiphertextException",
    "$fault": "client",
    "$metadata": {
        "httpStatusCode": 400,
        "requestId": "12345",
        "attempts": 1,
        "totalRetryDelay": 0
    },
    "__type": "InvalidCiphertextException",
    "message": "UnknownError",`
`}`

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @sudsbangs, have you taken a look to the following documentation:
- https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-email-sender.html#custom-email-sender-code-examples
- https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#symmetric-cmks

I hope this helps. Please let me know if you have any questions.

Thanks!

**Comment by ziaulrehman40** [comment]
So whatw as the issue?
